### ISSUE TITLE
`unnecessary_lambdas` false positive with `late final` variables

### ISSUE THREAD

**Issue by simolus3** [dart-lang/linter#2670]
**Describe the issue**

The unnecessary lambdas lint falsely flags usages of late final variables that may not be safe to replace with a tear off that gets evaluated right away.

**To Reproduce**

A common occurence of this may be the `test` package.

```dart
late final Client client;

setUpAll(() {
  client = Client();
});

tearDownAll(() => client.close()); // LINT
```

**Expected behavior**

I would not expect a lint on the last line, as replacing it with a tear-off causes an "unitialized late variable" error at runtime.

**Additional context**

I tested this on 2.13.0 stable and 2.14.0-115.0.dev.

**Comment by srawlins** [comment]
Closed by https://github.com/dart-lang/linter/pull/4089
